besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java

```diff
@@ -493,7 +509,10 @@ public BesuController build() {
             reorgLoggingThreshold,
             dataDirectory.toString());
 
-    final CachedMerkleTrieLoader cachedMerkleTrieLoader = new CachedMerkleTrieLoader(metricsSystem);
+    final CachedMerkleTrieLoader cachedMerkleTrieLoader =
+        besuComponent == null
```

Comment:
In what scenario would besuComponent be null here?

Reply:
Good catch. This is leftover from a prior revision where we were passing in the BesuComponent after construction time.